File: shared/src/commonMain/kotlin/com/xichen/cloudphoto/analytics/AppAnalyticsCatalog.kt
package com.xichen.cloudphoto.analytics

/**
 * Canonical `page` values for [com.xichen.cloudphoto.model.AppEventReportDto.page].
 */
object AnalyticsPages {
    const val PHOTO_TIMELINE = "photo_timeline"
    const val ALBUMS = "albums"
    const val CAMERA = "camera"
    const val STORAGE = "storage"
    const val STORAGE_ADD = "storage_add"
    const val STORAGE_EDIT = "storage_edit"
    const val STORAGE_TUTORIAL = "storage_tutorial"
    const val SETTINGS = "settings"
    const val PROFILE = "profile"
    const val ACCOUNT_SECURITY = "account_security"
    const val CHANGE_PASSWORD = "change_password"
    const val THEME_SETTINGS = "theme_settings"
    const val NOTIFICATION_SETTINGS = "notification_settings"
    const val HELP_FEEDBACK = "help_feedback"
    const val ABOUT = "about"
    const val OPEN_SOURCE_LICENSES = "open_source_licenses"
    const val LOGIN = "login"
    const val REGISTER = "register"
}

/**
 * Event identifiers sent as API `elementId` (文档与报表中的 eventId).
 *
 * Naming: `evt_` + domain + `_` + action[/target].
 */
object AnalyticsEventIds {
    const val BOTTOM_NAV_PHOTOS = "evt_bottom_nav_photos"
    const val BOTTOM_NAV_ALBUMS = "evt_bottom_nav_albums"
    const val BOTTOM_NAV_CAMERA = "evt_bottom_nav_camera"
    const val BOTTOM_NAV_STORAGE = "evt_bottom_nav_storage"
    const val BOTTOM_NAV_SETTINGS = "evt_bottom_nav_settings"

    const val LOGIN_SUBMIT = "evt_login_submit"
    const val LOGIN_GO_REGISTER = "evt_login_go_register"
    const val REGISTER_SUBMIT = "evt_register_submit"
    const val REGISTER_GO_LOGIN = "evt_register_go_login"

    const val PHOTO_SEARCH_TAP = "evt_photo_search_tap"
    const val PHOTO_GRID_ITEM = "evt_photo_grid_item"
    const val PHOTO_FULLSCREEN_CLOSE = "evt_photo_fullscreen_close"

    const val SETTINGS_PROFILE = "evt_settings_profile"
    const val SETTINGS_ACCOUNT_SECURITY = "evt_settings_account_security"
    const val SETTINGS_THEME = "evt_settings_theme"
    const val SETTINGS_NOTIFICATIONS = "evt_settings_notifications"
    const val NOTIFICATION_CLOUD_PUSH_TOGGLE = "evt_notification_cloud_push_toggle"
    const val SETTINGS_HELP = "evt_settings_help"
    const val FEEDBACK_SUBMIT = "evt_feedback_submit"
    const val SETTINGS_ABOUT = "evt_settings_about"
    const val ABOUT_UPLOAD_LOGS = "evt_about_upload_logs"
    const val ABOUT_OPEN_SOURCE_LICENSES = "evt_about_open_source_licenses"
    const val SETTINGS_LOGOUT = "evt_settings_logout_confirm"

    const val STORAGE_ADD_CONFIG = "evt_storage_add_config"
    const val STORAGE_TUTORIAL = "evt_storage_tutorial"
    const val STORAGE_EDIT_CONFIG = "evt_storage_edit_config"
}

File: iosApp/iosApp/Analytics/AppAnalyticsCatalog.swift
import Foundation

/// Analytics `page` / `elementId` 字面量，与 KMP `AppAnalyticsCatalog.kt`（`AnalyticsPages`、`AnalyticsEventIds`）及 Android 保持一致。
enum AppAnalyticsCatalog {

    enum Page {
        static let photoTimeline = "photo_timeline"
        static let albums = "albums"
        static let camera = "camera"
        static let storage = "storage"
        static let storageAdd = "storage_add"
        static let storageEdit = "storage_edit"
        static let storageTutorial = "storage_tutorial"
        static let settings = "settings"
        static let profile = "profile"
        static let accountSecurity = "account_security"
        static let changePassword = "change_password"
        static let themeSettings = "theme_settings"
        static let notificationSettings = "notification_settings"
        static let helpFeedback = "help_feedback"
        static let about = "about"
        static let openSourceLicenses = "open_source_licenses"
        static let login = "login"
        static let register = "register"
    }

    enum EventId {
        static let bottomNavPhotos = "evt_bottom_nav_photos"
        static let bottomNavAlbums = "evt_bottom_nav_albums"
        static let bottomNavCamera = "evt_bottom_nav_camera"
        static let bottomNavStorage = "evt_bottom_nav_storage"
        static let bottomNavSettings = "evt_bottom_nav_settings"
        static let loginSubmit = "evt_login_submit"
        static let loginGoRegister = "evt_login_go_register"
        static let registerSubmit = "evt_register_submit"
        static let registerGoLogin = "evt_register_go_login"
        static let photoSearchTap = "evt_photo_search_tap"
        static let photoGridItem = "evt_photo_grid_item"
        static let photoFullscreenClose = "evt_photo_fullscreen_close"
        static let settingsProfile = "evt_settings_profile"
        static let settingsAccountSecurity = "evt_settings_account_security"
        static let settingsTheme = "evt_settings_theme"
        static let settingsNotifications = "evt_settings_notifications"
        static let notificationCloudPushToggle = "evt_notification_cloud_push_toggle"
        static let settingsHelp = "evt_settings_help"
        static let feedbackSubmit = "evt_feedback_submit"
        static let settingsAbout = "evt_settings_about"
        static let aboutUploadLogs = "evt_about_upload_logs"
        static let aboutOpenSourceLicenses = "evt_about_open_source_licenses"
        static let settingsLogout = "evt_settings_logout_confirm"
        static let storageAddConfig = "evt_storage_add_config"
        static let storageTutorial = "evt_storage_tutorial"
        static let storageEditConfig = "evt_storage_edit_config"
    }

    /// 与 `MainTabView` 的 Tab 顺序、Android `Screen.mainTabRoutes` 一致。
    static let mainTabRoutes = ["photos", "albums", "camera", "storage", "settings"]

    /// 将主导航路由（如 `photos`）映射为上报用的 `page` 字段。
    static func page(forMainTabRoute route: String) -> String {
        switch route {
        case "photos": return Page.photoTimeline
        case "albums": return Page.albums
        case "camera": return Page.camera
        case "storage": return Page.storage
        case "settings": return Page.settings
        default:
            let normalized = route.replacingOccurrences(of: "/", with: "_")
            return normalized.isEmpty ? Page.photoTimeline : normalized
        }
    }
}
